src/main/java/org/opentripplanner/graph_builder/linking/SimpleStreetSplitter.java

```diff
@@ -154,13 +155,39 @@ else if (v instanceof BikeParkVertex)
         }
     }
 
-    /** Link this vertex into the graph to the closest walkable edge */
-    public boolean link (Vertex vertex) {
-        return link(vertex, TraverseMode.WALK, null);
+    /**
+     * Link this vertex into the graph to the closest walkable edge
+     * @param vertex The vertex to be linked.
+     * @param destructiveSplitting If true splitting is permanent (Used when linking transit stops etc.) when
+     *                             false Splitting is only for duration of a request. Since they are made from
+     *                             temporary vertices and edges.
+     * @return
+     */
+    public boolean linkToClosestWalkableEdge(Vertex vertex, final boolean destructiveSplitting) {
+        return linkToGraph(vertex, TraverseMode.WALK, null, destructiveSplitting);
+    }
+
+    public boolean linkToGraph(Vertex vertex, TraverseMode traverseMode, RoutingRequest options,
+                               final boolean destructiveSplitting) {
+        final TraverseModeSet traverseModeSet = new TraverseModeSet(traverseMode);
+        if (traverseMode == TraverseMode.BICYCLE) {
+            traverseModeSet.setWalk(true);
+        }
+        return linkToGraph(vertex, traverseModeSet, options, destructiveSplitting);
     }
 
-    /** Link this vertex into the graph */
-    public boolean link(Vertex vertex, TraverseMode traverseMode, RoutingRequest options) {
+    /**
+     *  Link this vertex into the graph
```

Comment:
This could benefit from an explanation of what it means to "link".

Reply:
Added some comments here: https://github.com/opentripplanner/OpenTripPlanner/pull/2762/commits/cf74d8132a221812ce2ed114a13c3faffd7cd282